packages/beacon-node/src/chain/archiver/index.ts

```diff
@@ -143,4 +174,159 @@ export class Archiver {
+
+    let expectedAttachedValidatorsProposalsCount = 0;
+    let finalizedAttachedValidatorsProposalsCount = 0;
+    let finalizedAttachedValidatorsOrphanCount = 0;
+    let finalizedAttachedValidatorsMissedCount = 0;
+
+    for (const checkpointHex of finalizedProposersCheckpoints) {
+      const checkpointState = checkpointStateCache.get(checkpointHex);
+
+      // Generate stats for attached validators if we have state info
+      if (checkpointState !== null) {
+        finalizedFoundCheckpointsInStateCache++;
+
+        const epochProposers = checkpointState.epochCtx.proposers;
+        const startSlot = checkpointState.epochCtx.epoch * SLOTS_PER_EPOCH;
+
+        for (let index = 0; index < epochProposers.length; index++) {
+          const slot = startSlot + index;
+
+          // Let skip processing the slots which are out of range
+          // Range to consider is:
+          //   lastFinalized.epoch * SLOTS_PER_EPOCH + 1, .... finalized.epoch * SLOTS_PER_EPOCH
+          if (slot <= lastFinalized.epoch * SLOTS_PER_EPOCH || slot > finalized.epoch * SLOTS_PER_EPOCH) {
+            continue;
+          }
+
+          const proposer = epochProposers[index];
+
+          // If this proposer was attached to this BN for this epoch
+          if (attachedProposers.includes(proposer)) {
+            expectedAttachedValidatorsProposalsCount++;
+
+            // Get what validators made canonical/non canonical proposals for this slot
+            const {canonicalVals, nonCanonicalVals} = slotProposers.get(slot) ?? {
+              canonicalVals: [],
+              nonCanonicalVals: [],
+            };
+            let wasFinalized = false;
+
+            if (canonicalVals.includes(proposer)) {
+              finalizedAttachedValidatorsProposalsCount++;
+              wasFinalized = true;
+            }
+            const attachedProposerNonCanSlotProposals = nonCanonicalVals.filter((nonCanVal) => nonCanVal === proposer);
+            finalizedAttachedValidatorsOrphanCount += attachedProposerNonCanSlotProposals.length;
+
+            // Check is this slot proposal was missed by this attached validator
+            if (!wasFinalized && attachedProposerNonCanSlotProposals.length === 0) {
+              finalizedAttachedValidatorsMissedCount++;
+            }
+          }
+        }
+      }
+    }
+
+    const attachedValidators: ProposalStats = {
+      expected: expectedAttachedValidatorsProposalsCount,
+      finalized: finalizedAttachedValidatorsProposalsCount,
+      orphaned: finalizedAttachedValidatorsOrphanCount,
+      missed: finalizedAttachedValidatorsMissedCount,
+    };
+
+    this.logger.info("All validators finalized proposal stats", {
+      ...allValidators,
+      finalizedCanonicalCheckpointsCount,
+      finalizedFoundCheckpointsInStateCache,
+    });
+    this.logger.info("Attached validators finalized proposal stats", {
+      ...attachedValidators,
+      finalizedAttachedValidatorsCount,
+    });
```

Comment:
Could just hide behind a feature flag, similar to validator monitor logs (to be added)
- https://github.com/ChainSafe/lodestar/issues/5336

Reply:
we can push all validators log (and finalizedAttachedValidatorsCount) to debug

Reply:
@g11tech  After seeing this log (`Attached validators finalized proposal stats expected=0, finalized=0, orphaned=0, missed=0, finalizedAttachedValidatorsCount=0`) on different beacon node instances I find it not very useful as a info log which should be printed out by default.

I think should reconsider and only enable it via flag as I suggested above. Right now, it seems really random as we do report stats on block proposals but not attestations. 

It also just prints out the proposal stats during which the beacon node was running, on restart, it will reset all values to 0. Unless someone is running hundreds of validators this log will not change for months.

Focusing the BN logs on just sync and peer info by default feels a lot cleaner to be as there might not even be validators attached.

Reply:
I do agree here

Reply:
We had a discussion about this in Discord here: https://discord.com/channels/593655374469660673/1105347927548907600/1110563872152236133

The compromise made is that this will not show on `info` if node is still syncing, there are no validators attached or if all stats are 0. There will also be a flag to disable this.